lib/perl/Genome/InstrumentData/AlignmentResult/Command/PicardRnaSeqMetrics.pm

```diff
@@ -78,8 +79,37 @@ sub chart_output_filename {
     return Genome::Model::Build::RnaSeq->default_chart_output_filename;
 }
 
+sub required_files_from_annotation_build {
+    (qw/ rRNA_MT_file annotation_file /);
+}
+
+sub does_annotation_build_have_required_files {
+    my ($self, $annotation_build, $reference_build) = Params::Validate::validate_pos(
+        @_, {isa => __PACKAGE__}, {type => OBJECT}, {type => OBJECT},
+    );
+
+    my @missing_files;
+    for my $file_method ( required_files_from_annotation_build() ) {
+        my $file = $annotation_build->$file_method('gtf', $reference_build->id ,0);
+        if ( not -s $file ) {
+            $file = $annotation_build->$file_method('gtf', undef, 0);
+            if ( not -s $file ) {
```

Comment:
This pattern of trying with the reference build and then without is now repeated thrice in this module.  Would it be worth pulling it out into its own subroutine?